alexandria/core/models.py

```diff
@@ -268,12 +269,18 @@ def set_content_vector(self):
             config=config,
             weight="B",
         )
+        self.save()
 
     def save(
         self, force_insert=False, force_update=False, using=None, update_fields=None
     ):
         self.set_checksum()
-        self.set_content_vector()
+        if (
+            settings.ALEXANDRIA_ENABLE_CONTENT_SEARCH
+            and self.variant == File.Variant.ORIGINAL
+            and not self.content_vector
+        ):
+            set_content_vector.delay_on_commit(self.pk)
```

Comment:
I'd move the logic into a `post_save` signal handler to trigger the appropriate Celery task(s)

Reply:
Done